src/maliput_osm/osm/osm_manager.cc

```diff
@@ -80,22 +118,65 @@ OSMManager::OSMManager(const std::string& osm_file_path, const ParserConfig& con
       }
     }
   }
+
+  // Organize segments into junctions.
+  std::vector<std::unordered_set<std::string>> potential_junction_lanes_set;
+  // Group lanes that belong to a complex junction.
+  for (const auto& segment : segments) {
+    for (const auto& lane : segment.second.lanes) {
+      const bool complex_junction_in_predecessor = IsPotentialComplexJunction(lane.predecessors);
+      const bool complex_junction_in_successor = IsPotentialComplexJunction(lane.successors);
+
+      if (complex_junction_in_predecessor) {
+        // check the lanes in the predecessor and add them to the potential_junction_lanes_set
+        AddLanesToCollectionOfSet(lane.predecessors, potential_junction_lanes_set);
+      }
+      if (complex_junction_in_successor) {
+        // check the lanes in the successor and add them to the potential_junction_lanes_set
+        AddLanesToCollectionOfSet(lane.successors, potential_junction_lanes_set);
+      }
+    }
+  }
+
+  // Create junctions for complex connections
+  for (const auto& set : potential_junction_lanes_set) {
+    Junction junction;
+    Junction::Id junction_id{""};
+    MALIPUT_THROW_UNLESS(!set.empty());
```

Comment:
Is this assertion necessary? Can we make it somewhere else?

Reply:
It's not. By implementation, this will never be empty.
Done.